deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/CustomLayer/GradientCheckTest.java

```diff
@@ -0,0 +1,97 @@
+package org.deeplearning4j.nn.layers.CustomLayer;
+
+
+import org.deeplearning4j.nn.api.OptimizationAlgorithm;
+import org.deeplearning4j.nn.conf.ComputationGraphConfiguration;
+import org.deeplearning4j.nn.conf.NeuralNetConfiguration;
+import org.deeplearning4j.nn.conf.distribution.UniformDistribution;
+import org.deeplearning4j.nn.conf.layers.LossLayer;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.junit.Test;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.learning.config.NoOp;
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.deeplearning4j.gradientcheck.GradientCheckUtil.checkGradients;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * created by jingshu
+ */
+
+public class GradientCheckTest {
+    private static final boolean PRINT_RESULTS = true;
+    private static final boolean RETURN_ON_FIRST_FAILURE = false;
+    private static final double DEFAULT_EPS = 1e-6;
+    private static final double DEFAULT_MAX_REL_ERROR = 1e-3;
+    private static final double DEFAULT_MIN_ABS_ERROR = 1e-8;
+
+    private static Logger log = (Logger) LoggerFactory.getLogger(GradientCheckTest.class);
+
+    @Test
+    public void elementWiseMultiplicationLayerTest(){
```

Comment:
This gradient check is a god start - but is insufficient. To properly test it, we require a layer below it (otherwise the epsilonNext component could be incorrect, and we would never know from this test). Something like a DenseLayer below it should be fine for that.